src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -96,6 +97,11 @@ public void acquireLease() throws ConnectException {
     }
   }
 
+  private void reset() {
```

Comment:
Instead of adding a new method, could we just do this in the `close()` method in a finally block so that we attempt to close the reader/writer gracefully first before nulling them out? Then replace calls to `reset()` with `close()` in this file instead?